iot_config/src/route_service.rs

```diff
@@ -0,0 +1,200 @@
+use crate::{
+    lora_field::{DevAddrRange, Eui},
+    org::OrgStatus,
+    route::{self, Route},
+    GrpcResult, GrpcStreamResult, Settings,
+};
+use anyhow::Result;
+use futures::stream::StreamExt;
+use helium_proto::services::iot_config::{
+    self, ActionV1, RouteCreateReqV1, RouteDeleteReqV1, RouteDevaddrsReqV1, RouteDevaddrsResV1,
+    RouteEuisReqV1, RouteEuisResV1, RouteGetReqV1, RouteListReqV1, RouteListResV1,
+    RouteStreamReqV1, RouteStreamResV1, RouteUpdateReqV1, RouteV1,
+};
+use sqlx::{Pool, Postgres};
+use std::sync::Arc;
+use tokio::{
+    pin,
+    sync::broadcast::{Receiver, Sender},
+};
+use tonic::{Request, Response, Status};
+
+pub struct RouteService {
+    pool: Pool<Postgres>,
+    update_channel: Arc<Sender<RouteStreamResV1>>,
+}
+
+impl RouteService {
+    pub async fn new(settings: &Settings) -> Result<Self> {
+        let (update_tx, _) = tokio::sync::broadcast::channel(128);
+
+        Ok(Self {
+            pool: settings.database.connect(10).await?,
+            update_channel: Arc::new(update_tx),
+        })
+    }
+
+    fn subscribe_to_routes(&self) -> Receiver<RouteStreamResV1> {
+        self.update_channel.subscribe()
+    }
+}
+
+#[tonic::async_trait]
+impl iot_config::Route for RouteService {
+    async fn list(&self, request: Request<RouteListReqV1>) -> GrpcResult<RouteListResV1> {
+        let request = request.into_inner();
+
+        let proto_routes: Vec<RouteV1> = route::list_routes(request.oui, &self.pool)
+            .await
+            .map_err(|_| Status::internal("route list failed"))?
```

Comment:
i notice all error response are of `internal` type.  Should we not be more descriptive where possible ?  For example if the user attempts to list or get an invalid route, maybe a not found response

Reply:
Yes, they should probably be more descriptive and specific

Reply:
so i went back over them and made a few updates but for the most part these all amount to "something went wrong in the db update" so i could be a little more explicit in that regard with the message, but i wouldn't have a better suggestion for which _type_ of status error each one should have than the current ones (options defined here: https://docs.rs/tonic/latest/tonic/struct.Status.html); i'm open to arguments for/against and in particular.

i also think as the API is fleshed out in subsequent changes we'll have alternative failure scenarios like `unauthorized` that just aren't relevant with this initial build